tests/tcl/tests/unit/expire.tcl

```diff
@@ -1,206 +0,0 @@
-        # server is under pressure, so if it does not work give it a few more
-        # chances.
-        for {set j 0} {$j < 3} {incr j} {
-            r del x
-            r setex x 1 somevalue
-            after 900
-            set a [r get x]
-            after 1100
-            set b [r get x]
-            if {$a eq {somevalue} && $b eq {}} break
-        }
-        list $a $b
-    } {somevalue {}}
-
-    test {PEXPIRE/PSETEX/PEXPIREAT can set sub-second expires} {
-        # This test is very likely to do a false positive if the
-        # server is under pressure, so if it does not work give it a few more
-        # chances.
-        for {set j 0} {$j < 3} {incr j} {
-            r del x y z
-            r psetex x 100 somevalue
-            after 80
-            set a [r get x]
-            after 2100
-            set b [r get x]
-
-            r set x somevalue
-            r pexpire x 100
-            after 80
-            set c [r get x]
-            after 2100
-            set d [r get x]
-
-            r set x somevalue
-            r pexpireat x [expr ([clock seconds]*1000)+100]
-            after 80
-            set e [r get x]
-            after 2100
-            set f [r get x]
-
-            if {$a eq {somevalue} && $b eq {} &&
-                $c eq {somevalue} && $d eq {} &&
-                $e eq {somevalue} && $f eq {}} break
-        }
-        list $a $b
-    } {somevalue {}}
-
-    test {TTL returns tiem to live in seconds} {
-        r del x
-        r setex x 10 somevalue
-        set ttl [r ttl x]
-        assert {$ttl > 8 && $ttl <= 10}
-    }
-
-    test {PTTL returns time to live in milliseconds} {
-        r del x
-        r setex x 1 somevalue
-        set ttl [r pttl x]
-        assert {$ttl > 900 && $ttl <= 1000}
-    }
-
-    test {TTL / PTTL return -1 if key has no expire} {
-        r del x
-        r set x hello
-        list [r ttl x] [r pttl x]
-    } {-1 -1}
-
-    test {TTL / PTTL return -2 if key does not exit} {
-        r del x
-        list [r ttl x] [r pttl x]
-    } {-2 -2}
-
-    test {Redis should actively expire keys incrementally} {
-        r flushdb
-        r psetex key1 500 a
-        r psetex key2 500 a
-        r psetex key3 500 a
-        r dbsize scan
-        after 100
-        set size1 [r dbsize]
-        # Redis expires random keys ten times every second so we are
-        # fairly sure that all the three keys should be evicted after
-        # one second.
-        after 2000
-        r dbsize scan
-        after 100
-        set size2 [r dbsize]
-        list $size1 $size2
-    } {3 0}
-
-    test {5 keys in, 5 keys out} {
-        r flushdb
-        r set a c
-        r expire a 5
-        r set t c
-        r set e c
-        r set s c
-        r set foo b
-        lsort [r keys *]
-    } {a e foo s t}
-
-    test {EXPIRE with empty string as TTL should report an error} {
-        r set foo bar
-        catch {r expire foo ""} e
-        set e
-    } {*not an integer*}
```

Comment:
is there corresponding go code for this case?

Reply:
At first I thought that the `go-redis Expire` method would have type checking, so it was not implemented, it has been added